### PR TITLE
Add alias GVL IDs tests

### PR DESCRIPTION
**Pull request by marki1an**
Tests for - [1835](https://github.com/prebid/prebid-server-java/pull/1835)

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 1]
```diff
@@ -212,4 +212,33 @@ class BidValidationSpec extends BaseSpec {
         def metrics = defaultPbsService.sendCollectedMetricsRequest()
         assert metrics["adapter.generic.requests.bid_validation"] == initialMetricValue + 1
     }
+
+    def "PBS should return error invalid request format request.ext.aliasgvild unknown bidder alias"() {
```
Comment (hupaloo) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:216]: Not clear name. Use pattern "Should ExpectedBehavior When StateUnderTest" for test naming

File: src/test/groovy/org/prebid/server/functional/model/request/auction/Prebid.groovy [hunk 2]
```diff
@@ -16,6 +16,7 @@ class Prebid {
     Amp amp
     Channel channel
     Map<String, BidderName> aliases
+    Map<String, Integer> aliasgvlids
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/model/request/auction/Prebid.groovy:19]: Please use camelCase for fields. To properly map different styles, use Jackson annotations.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 3]
```diff
@@ -212,4 +212,34 @@ class BidValidationSpec extends BaseSpec {
         def metrics = defaultPbsService.sendCollectedMetricsRequest()
         assert metrics["adapter.generic.requests.bid_validation"] == initialMetricValue + 1
     }
+
+    def "PBS should return error when between refers to unknown bidder alias"() {
+        given: "Default basic BidRequest"
+        def bidRequest = BidRequest.defaultBidRequest
+        bidRequest.ext.prebid.aliasgvlids = ["between": 1]
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:219]: What does `between` signify? Is it a bidder name? Or can it be any string?

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 4]
```diff
@@ -212,4 +212,34 @@ class BidValidationSpec extends BaseSpec {
...
+        bidRequest.ext.prebid.aliases = ["appnexus": GENERIC]
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:220]: 1) Please try to not use real names in tests.
2) How `appnexus` ties to `between`?

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 5]
```diff
@@ -212,4 +212,34 @@ class BidValidationSpec extends BaseSpec {
...
+
+        when: "Sending auction request to PBS"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Request should fail with error"
+        def exception = thrown(PrebidServerException)
+        assert exception.responseBody.contains("Invalid request format: request.ext.prebid.aliasgvlids. " +
+                "vendorId between refers to unknown bidder alias: 1")
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:228]: 1) Again `between`. What does it signify here? Why it's not a reference to existing variable if it's the same thing from the `given` section?
2) What is 1?

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 6]
```diff
@@ -212,4 +212,34 @@ class BidValidationSpec extends BaseSpec {
         def metrics = defaultPbsService.sendCollectedMetricsRequest()
         assert metrics["adapter.generic.requests.bid_validation"] == initialMetricValue + 1
     }
+
+    def "PBS should return error when between refers to unknown bidder alias"() {
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:216]: Misleading test naming. What does `between` means? Looks like test should be named `PBS should return an error when GVL Id alias refers to unknown bidder alias`.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 7]
```diff
@@ -212,4 +212,34 @@ class BidValidationSpec extends BaseSpec {
...
+        when: "Sending auction request to PBS"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Request should fail with error"
+        def exception = thrown(PrebidServerException)
+        assert exception.responseBody.contains("Invalid request format: request.ext.prebid.aliasgvlids. " +
+                "vendorId between refers to unknown bidder alias: 1")
+    }
+
+    def "PBS should return error when aliasgvlids value lowest that one"() {
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:231]: `PBS should return an error when GVL ID alias  value is lower that one`

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 8]
```diff
@@ -212,4 +212,34 @@ class BidValidationSpec extends BaseSpec {
...
+        given: "Default basic BidRequest"
+        def bidRequest = BidRequest.defaultBidRequest
+        bidRequest.ext.prebid.aliasgvlids = ["appnexus": 0]
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:234]: 1) Please try to not use real names in tests.
2) Test can also be parametrized by trying to pass in `-1`.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 9]
```diff
@@ -212,4 +212,34 @@ class BidValidationSpec extends BaseSpec {
...
+        bidRequest.ext.prebid.aliases = ["appnexus": GENERIC]
+
+        when: "Sending auction request to PBS"
+        defaultPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Request should fail with error"
+        def exception = thrown(PrebidServerException)
+        assert exception.responseBody.contains("Invalid request format: request.ext.prebid.aliasgvlids. " +
+                "Invalid vendorId appnexus for alias: 0. Choose a different vendorId, or remove this entry.")
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:243]: Please don't hardcode names. Use string interpolation and refer to values from the `given` section.

**Review by Net-burst — DISMISSED** [review]

File: src/test/groovy/org/prebid/server/functional/model/request/auction/Prebid.groovy (hunk 2 above)
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/model/request/auction/Prebid.groovy:19]: I think it's not necessary, because in ExtRequestPrebid the same naming https://github.com/prebid/prebid-server-java/blob/8ecf10089c1c2b6a5f9ef0c6f6ad60042fd5716f/src/main/java/org/prebid/server/proto/openrtb/ext/request/ExtRequestPrebid.java#L40

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy (hunk 3 above)
Comment (hupaloo) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:219]: First of all, do not resolve not yours comments.
That comment wasn't answered here nor resolved in the code.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy (hunk 4 above)
Comment (hupaloo) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:220]: Same, that comment wasn't answered here nor resolved in the code.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy (hunk 5 above)
Comment (hupaloo) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:228]: Same, that comment wasn't answered here nor resolved in the code.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy (hunk 6 above)
Comment (hupaloo) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:216]: Same, that comment wasn't answered here nor resolved in the code.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy (hunk 7 above)
Comment (hupaloo) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:231]: Same, that comment wasn't answered here nor resolved in the code.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy (hunk 8 above)
Comment (hupaloo) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:234]: Same, that comment wasn't answered here nor resolved in the code.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy (hunk 9 above)
Comment (hupaloo) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:243]: Same, that comment wasn't answered here nor resolved in the code.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy (hunk 3 above)
Comment (hupaloo) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:219]: If that is any bidder name, so then use random string generator from PBSUtils to show that test is not bidder specific.

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy (hunk 4 above)
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:220]: 1. Why don't use real names in tests?
We already used real names :)
2. Nothing ties them.

**Review by hupaloo — COMMENTED** [review]
A couple of minor remarks and we are ready to go

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 10]
```diff
@@ -212,4 +212,40 @@ class BidValidationSpec extends BaseSpec {
         def metrics = defaultPbsService.sendCollectedMetricsRequest()
         assert metrics["adapter.generic.requests.bid_validation"] == initialMetricValue + 1
     }
+
+    def "PBS should return an error when GVL Id alias refers to unknown bidder alias"() {
+        given: "Default basic BidRequest"
```
[the remark this answers is not in the document]
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy]: Done

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 11]
```diff
@@ -212,4 +212,40 @@ class BidValidationSpec extends BaseSpec {
...
+        def bidderName = PBSUtils.randomString
+        def validId = 1;
```
[the remark this answers is not in the document]
Comment (marki1an) [reply src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy]: Done

File: src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy [hunk 12]
```diff
@@ -214,9 +214,9 @@ class BidValidationSpec extends BaseSpec {
     }
 
     def "PBS should return an error when GVL Id alias refers to unknown bidder alias"() {
-        given: "Default basic BidRequest"
+        given: "Default basic BidRequest with aliasgvlids and aliases"
         def bidderName = PBSUtils.randomString
-        def validId = 1;
+        def validId = PBSUtils.getRandomNumber(1)
```
Comment (hupaloo) [inline src/test/groovy/org/prebid/server/functional/tests/BidValidationSpec.groovy:219]: Well, take a look at a boundary value analysis and equivalence partitioning testing.
'validId = 1' was good here as it is on the bound.
With a random number from getRandomNumber() you don't verify that just exactly after 0 integer PBS processes GVL Id alias in a different way.

**Review by hupaloo — APPROVED** [review]
